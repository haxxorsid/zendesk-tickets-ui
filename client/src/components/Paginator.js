import React from 'react';

const Paginator = (props) => (
  <ul className="paginator">
    { props.numberOfPages &&
      props.numberOfPages.map(page => (
        (
          <li 
            key={page}
            className={page === props.currentPage ? "current-page paginator-item" : "paginator-item"}
            onClick={() => props.onPageChange(page)}
            >{page + 1}
          </li>
        )
      ))
    }
  </ul>
)

export default Paginator;