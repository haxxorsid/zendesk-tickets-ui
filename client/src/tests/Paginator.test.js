import React from 'react';
import renderer from 'react-test-renderer'
import Paginator from '../components/Paginator';

test('Should render PageSelector correctly with props', () => {
    const numberOfPages = [1, 2, 3, 4]
    const tree = renderer
        .create(<Paginator currentPage={1} numberOfPages={numberOfPages} />)
        .toJSON();
    expect(tree).toMatchSnapshot()
})

test('Should change current page on click', () => {
    const numberOfPages = [1, 2, 3, 4]
    const onPageChangeFn = jest.fn()
    const component = renderer
        .create(<Paginator currentPage={1} numberOfPages={numberOfPages} onPageChange={onPageChangeFn}/>)
    component.root.findAllByType('li')[1].props.onClick()
    expect(onPageChangeFn).toHaveBeenCalledWith(2)
})