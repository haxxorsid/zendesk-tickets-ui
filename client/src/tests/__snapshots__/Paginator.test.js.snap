// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render PageSelector correctly with props 1`] = `
<ul
  className="paginator"
>
  <li
    className="current-page paginator-item"
    onClick={[Function]}
  >
    2
  </li>
  <li
    className="paginator-item"
    onClick={[Function]}
  >
    3
  </li>
  <li
    className="paginator-item"
    onClick={[Function]}
  >
    4
  </li>
  <li
    className="paginator-item"
    onClick={[Function]}
  >
    5
  </li>
</ul>
`;
